ql/src/java/org/apache/hadoop/hive/ql/ddl/table/execute/AlterTableExecuteAnalyzer.java

```diff
@@ -69,6 +70,7 @@ public AlterTableExecuteAnalyzer(QueryState queryState) throws SemanticException
   protected void analyzeCommand(TableName tableName, Map<String, String> partitionSpec, ASTNode command)
       throws SemanticException {
     Table table = getTable(tableName);
+    DDLUtils.validateTableIsIceberg(table);
```

Comment:
As of today these API's are supported only for iceberg but aren't limited, shouldn't restrict or hardcode iceberg here.